src/crimson/os/seastore/cache.cc

```diff
@@ -1799,17 +1839,26 @@ Cache::replay_delta(
 	DEBUG("replay extent is not present, so delta is obsolete at {} {} -- {}",
 	      journal_seq, record_base, delta);
 	assert(delta.pversion > 0);
-	return replay_delta_ertr::make_ready_future<bool>(true);
+	return replay_delta_ertr::make_ready_future<std::pair<bool, CachedExtentRef>>(
+	  std::make_pair(false, nullptr));
```

Comment:
Is it intentional/related to change to false here ?

Reply:
Yes, if we return true here, the caller (e.g., CircularBoundedJounal) will consider that the extent is replayed correctly (even though the extent is obsolete). Do you think 	
```
return replay_delta_ertr::make_ready_future<std::pair<bool, CachedExtentRef>>(
	  std::make_pair(true, nullptr));
```
is more accurate?

Reply:
I think so, and it would be nice to split the fixes in separate commits.

Reply:
Done.

Reply:
Oh, sorry about the confusion. I meant it seems to be reasonable to change to false here, but would be nice to split it out to a different commit as a separate fix.

I'd be OK with either true/false in this PR, it doesn't seem like a critical issue.